test/internal_rules.jl

```diff
@@ -379,12 +312,91 @@ end
 
         A = [1.3 0.5; 0.5 1.5]
         b = [1., 2.]
-        V = [1.0 0.0; 0.0 0.0]
         dA = zero(A)
         Enzyme.autodiff(Reverse, h, Active, Duplicated(A, dA), Const(b))
 
         dA_sym = - (transpose(A) \ [1.0, 0.0]) * transpose(A \ b)
-        @test isapprox(dA, dA_sym)
+        @test isapprox((dA + dA') / 2, (dA_sym + dA_sym') / 2)
+    end
+    @testset "Unit test for `cholesky` (regression test for #1307)" begin
+        # This test checks the `cholesky` rules without involving `ldiv!`
+        function f(A)
+            C = cholesky(A * adjoint(A))
+            return sum(abs2, C.L * C.U)
+        end
+        @testset for TE in (Float64, ComplexF64)
+            A = rand(TE, 3, 3)
+            test_forward(f, Duplicated, (A, Duplicated))
+            test_reverse(f, Active, (A, Duplicated))
+            @testset "Compare against function bypassing `cholesky`" begin
+                g(A) = sum(abs2, A * adjoint(A))
+                # If C = cholesky(A * A'), we have A * A' ≈ C.L * C.U, so `g`
+                # is essentially the same function as `f`, but bypassing `cholesky`.
+                # We can therefore use this to check that we get the correct derivatives.
+                @testset "Without wrapper" begin
+                    @testset "Forward mode" begin
+                        dA = rand(TE, size(A)...)
+                        d1 = autodiff(Forward, f, Duplicated, Duplicated(A, dA))
+                        d2 = autodiff(Forward, g, Duplicated, Duplicated(A, dA))
+                        @test all(d1 .≈ d2)
+                    end
+
+                    @testset "Reverse mode" begin
+                        dA1 = zero(A)
+                        dA2 = zero(A)
+                        autodiff(Reverse, f, Active, Duplicated(A, dA1))
+                        autodiff(Reverse, g, Active, Duplicated(A, dA2))
+                        @test dA1 ≈ dA2
+                    end
+                end
+                if TE == Float64
```

Comment:
Same, I would simplify these to just call `test_forward` and `test_reverse` using the combinations `[(Symmetric, Float64), (Hermitian, Float64), (Hermitian, ComplexF64)]`.

Reply:
That doesn't work IIRC because of issues with FiniteDiff.